source/common/http/utility.cc

```diff
@@ -562,5 +563,51 @@ void Utility::traversePerFilterConfigGeneric(
   }
 }
 
+std::string Utility::PercentEncoding::encode(absl::string_view value) {
+  std::string encoded;
+  for (size_t i = 0; i < value.size(); ++i) {
+    const char& ch = value[i];
+    // Unreserved characters. unreserved = ALPHA / DIGIT / "-" / "." / "_" / "~".
+    // https://tools.ietf.org/html/rfc3986#section-2.3.
+    if (absl::ascii_isalnum(ch) || ch == '-' || ch == '_' || ch == '.' || ch == '~') {
+      encoded.push_back(ch);
+      // TODO(dio): check for absl::is_space(ch), to encode it as '+' if desired.
+    } else {
+      // For consistency, URI producers should use uppercase hexadecimal digits for all
+      // percent-encodings. https://tools.ietf.org/html/rfc3986#section-2.1.
+      absl::StrAppend(&encoded, fmt::format("%{:02X}", ch));
+    }
+  }
+  return encoded;
+}
+
+std::string Utility::PercentEncoding::decode(absl::string_view encoded) {
```

Comment:
Is this used anywhere?

Reply:
This is for testing purpose only. Should I move this as a test helper?